test/core/linter_test.py

```diff
@@ -371,27 +372,52 @@ def test__linter__encoding(fname, config_encoding, lexerror):
     assert lexerror == (SQLLexError in [type(v) for v in result.get_violations()])
 
 
+# noqa tests require a rule_set, therefore we construct dummy rule set for glob matching.
+dummy_rule_set = [BaseRule(f"L{i:03d}", "dummy_rule") for i in range(1, 100)]
+
+
 @pytest.mark.parametrize(
-    "input,expected",
+    "input,expected,rule_set",
     [
-        ("", None),
-        ("noqa", NoQaDirective(0, None, None)),
-        ("noqa?", SQLParseError),
-        ("noqa:", NoQaDirective(0, None, None)),
-        ("noqa:L001,L002", NoQaDirective(0, ("L001", "L002"), None)),
-        ("noqa: enable=L005", NoQaDirective(0, ("L005",), "enable")),
-        ("noqa: disable=L010", NoQaDirective(0, ("L010",), "disable")),
-        ("noqa: disable=all", NoQaDirective(0, None, "disable")),
-        ("noqa: disable", SQLParseError),
+        ("", None, dummy_rule_set),
```

Comment:
Nitpick: It appears that all the cases of this test (`test_parse_noqa `) use the same rule set. If so, should we consider omitting it from the fixture and just using `dummy_rule_set` inside the test function? Or do we expect that future tests may use something other than `dummy_rule_set`?

Reply:
yeah I agree, I did actually initially do it like that but writing functions that reference global variables never sits well with me so I included it as an explicit argument to the function. tbh you're right though because it's just a test case, not a function that would ever get called elsewhere. Force of habit on my behalf 😛